include/mbedtls/ssl.h

```diff
@@ -1153,6 +1155,8 @@ struct mbedtls_ssl_config
     const mbedtls_ecp_group_id *MBEDTLS_PRIVATE(curve_list); /*!< allowed curves             */
 #endif
 
+    uint16_t MBEDTLS_PRIVATE(supported_groups[MBEDTLS_SSL_MAX_NAMED_GROUPS]);
```

Comment:
I think it should be `MBEDTLS_PRIVATE(supported_groups)[...]`?

Reply:
After discussion with @mpg @brett-warren-arm, I think we want a user-provided list here.